api/core/tools/tool_manager.py

```diff
@@ -201,7 +201,7 @@ def _init_runtime_parameter(cls, parameter_rule: ToolParameter, parameters: dict
             init runtime parameter
         """
         parameter_value = parameters.get(parameter_rule.name)
-        if not parameter_value:
+        if not parameter_value and parameter_value != 0:
```

Comment:
what about change it to `if parameter_value == None` ?

Reply:
actually, I can't reproduce all the false value of parameter_value here, cause of the frontend also has validation.
I see the `ToolParameterType` has string,boolean,select,file,  maybe it is an empty string or None here?  so I only add `!=0` here.

Reply:
Got it, frontend has already ensured all parameters are in correct type, validation here need only to check if it's empty, SGTM